File: samples/Test/Classes/Control/ActivityIndicatorViewTest.cpp

#include "ActivityIndicatorViewTest.h"

ActivityIndicatorViewTest::ActivityIndicatorViewTest()
{
    this->setTitle("CAActivityIndicatorView");
}

ActivityIndicatorViewTest::~ActivityIndicatorViewTest()
{

}

void ActivityIndicatorViewTest::viewDidLoad()
{
    this->getView()->setColor(CAColor4B::GRAY);
    
    auto idc1 = CAActivityIndicatorView::createWithLayout(DLayout(DHorizontalLayoutFill, DVerticalLayout_NH_C(0.2f, 1/6.f)));
    idc1->setStyle(CAActivityIndicatorView::Style::WhiteLarge);
    this->getView()->addSubview(idc1);
    
    auto idc2 = CAActivityIndicatorView::createWithLayout(DLayout(DHorizontalLayoutFill, DVerticalLayout_NH_C(0.2f, 2/6.f)));
    idc2->setStyle(CAActivityIndicatorView::Style::GrayLarge);
    this->getView()->addSubview(idc2);
    
    auto idc3 = CAActivityIndicatorView::createWithLayout(DLayout(DHorizontalLayoutFill, DVerticalLayout_NH_C(0.2f, 3/6.f)));
    idc3->setStyle(CAActivityIndicatorView::Style::White);
    this->getView()->addSubview(idc3);
    
    auto idc4 = CAActivityIndicatorView::createWithLayout(DLayout(DHorizontalLayoutFill, DVerticalLayout_NH_C(0.2f, 4/6.f)));
    idc4->setStyle(CAActivityIndicatorView::Style::Gray);
    this->getView()->addSubview(idc4);
    
    auto idc5 = CAActivityIndicatorView::createWithLayout(DLayout(DHorizontalLayoutFill, DVerticalLayout_NH_C(0.2f, 5/6.f)));
    idc5->setStyle(CAActivityIndicatorView::Style::Image);
    this->getView()->addSubview(idc5);
    
    CAImage* image = CAImage::create("image/indicator2.png");
    CAImageView* imageView = CAImageView::createWithLayout(DLayout(DHorizontalLayout_L_R(100, 100), DVerticalLayout_T_B(100,100)));
    imageView->setFrame(DRect(0, 0, 100, 100));
    imageView->setImage(image);
    
    idc5->setActivityIndicatorView(imageView);
}

void ActivityIndicatorViewTest::viewDidUnload()
{
    // Release any retained subviews of the main view.
    // e.g. self.myOutlet = nil;
}

File: samples/Test_JS/Resources/js/Control/ActivityIndicatorViewTest.js

var ActivityIndicatorViewTest = ca.CAViewController.extend({
    ctor: function (num) {
        this._super();
    },
    viewDidLoad: function() {

        var idc1 = ca.CAActivityIndicatorView.createWithLayout(ca.DLayout.set(ca.DHorizontalLayout.fill(), ca.DVerticalLayout_NH_C(0.2, 0.167)));
        idc1.setStyle(ca.CAActivityIndicatorView.Style.WhiteLarge);
        idc1.startAnimating();
        this.getView().addSubview(idc1);


        var idc2 = ca.CAActivityIndicatorView.createWithLayout(ca.DLayout.set(ca.DHorizontalLayout.fill(), ca.DVerticalLayout_NH_C(0.2, 0.334)));
        idc2.setStyle(ca.CAActivityIndicatorView.Style.GrayLarge);
        idc2.startAnimating();
        this.getView().addSubview(idc2);


        var idc3 = ca.CAActivityIndicatorView.createWithLayout(ca.DLayout.set(ca.DHorizontalLayout.fill(), ca.DVerticalLayout_NH_C(0.2, 0.5)));
        idc3.setStyle(ca.CAActivityIndicatorView.Style.White);
        idc3.setCycleTime(1.0);
        idc3.setTimesOneCycle(12);
        idc3.startAnimating();
        this.getView().addSubview(idc3);


        var idc4 = ca.CAActivityIndicatorView.createWithLayout(ca.DLayout.set(ca.DHorizontalLayout.fill(), ca.DVerticalLayout_NH_C(0.2, 0.667)));
        idc4.setStyle(ca.CAActivityIndicatorView.Style.Gray);
        idc4.setCycleTime(1.0);
        idc4.setTimesOneCycle(12);
        idc4.startAnimating();
        this.getView().addSubview(idc4);


        var imageView = ca.CAImageView.createWithLayout(ca.DLayout.set(ca.DHorizontalLayout_L_R(100, 100), ca.DVerticalLayout_T_B(100,100)));
        imageView.setFrame(ca.DRect.set(0, 0, 100, 100));
        imageView.setImage(ca.CAImage.create("image/indicator2.png"));

        var idc5 = ca.CAActivityIndicatorView.createWithLayout(ca.DLayout.set(ca.DHorizontalLayout.fill(), ca.DVerticalLayout_NH_C(0.2, 0.833)));
        idc5.setStyle(ca.CAActivityIndicatorView.Style.Image);
        idc5.setActivityIndicatorView(imageView);
        idc5.startAnimating();
        this.getView().addSubview(idc5);
    },
});
